nestkernel/recording_device.cpp

```diff
@@ -226,6 +229,8 @@ nest::RecordingDevice::Parameters_::set( const RecordingDevice& rd,
   updateValue< bool >( d, names::flush_records, flush_records_ );
   updateValue< bool >( d, names::close_on_reset, close_on_reset_ );
 
+  updateValue< bool >( d, names::use_gid, use_gid_ );
```

Comment:
This line requires to assign to a temporary first and only after checking if a label is set put it in the global variable. If `use_gid` is set to `true` by the user and no label is set, a `BadProperty` exception should be thrown with an appropriate error message.